torch/csrc/distributed/c10d/symm_mem/nvshmem_extension.cu

```diff
@@ -28,12 +28,20 @@ namespace c10d::nvshmem_extension {
 
 constexpr int MiB = 1024 * 1024;
 
+extern "C" void nvshmem_init() __attribute__((weak));
```

Comment:
Wait why do we load the first statically? Seems like a bug that just increases binary size of the wheel?

Reply:
@Skylion007 `libnvshmem_device.a` needs to be statically linked AFAIK -- it is a device-side lib that mainly stores NVSHMEM's device-side global states, and there is no .so alternative. It is relatively lightweight tho -- 6.8 MB.